### DISCUSSION TITLE
Help Wanted!

### DISCUSSION THREAD

**Discussion by gusmanb** [gusmanb/logicanalyzer#127] [Announcements]
Hi all!

I'm preparing something that is going to increase the number of protocol analyzers massively, a bridge to allow the usage of Sigrok decoders 😄 

The base concept is already working, the system is capable of loading the decoders, execute the analysis and represent the data, is still basic but it works. Here is a small example of the SPI decoder.

![imagen](https://github.com/user-attachments/assets/5a5daefd-f4f3-457d-979d-dc4f1aac077d)

As you can see in the image the number of decoders supported is really high compared to the original ones, basically all the non-stacked decoders are supported (in the future I might also add support for the stacked ones, but let's go step by step 😃 )

I have created the bridge from the ground up just using the documentation, there is no code reused from Sigrok and thus, I surely have errors on the implementation.

And there is where I need your help.

The decoders for what I have data do work, but I would need data from more protocols/devices to test as many decoders as possible.
If you can provide a capture for any of these protocol/devices please, upload it to this thread, I will update the list of missing ones as the people provides them. 

This is the list of decoders and the status of the data requirement:

|Protocol/Device|Status|Provided by|Notes|
|---|---|---|---|
|Audio Codec '97|Missing|||
|Aosong AM230x/DHTxx/RHTxx|Provided|[fbgentle1](https://github.com/fbgentle1)||
|Advanced User Debugger|Missing|||
|Atmel Program and Debug Interface|Missing|||
|Digital calipers|Missing|||
|Controller Area Network|Missing|||
|HDMI-CEC|Missing|||
|Compact Joint Test Action Group (IEEE 1149.7)|Missing|||
|Edge counter|Provided|[Dr. Gusman](https://github.com/gusmanb)||
|Digital Addressable Lighting Interface|Missing|||
|DCF77 time protocol|Missing|||
|Digital Serial Interface|Missing|||
|RFID EM4100|Missing|||
|RFID EM4205/EM4305|Missing|||
|FlexRay|Missing|||
|Gray code and rotary encoder|Missing|||
|Guess bitrate/baudrate|Missing|||
|Inter-Integrated Circuit|Provided|[Dr. Gusman](https://github.com/gusmanb)|Tested with 7 bit addresses, 10 bit capture required|
|Integrated Interchip Sound|Provided|[fbgentle1](https://github.com/fbgentle1)||
|IEEE-488 GPIB/HPIB/IEC|Missing|||
|IR IRMP|Missing|||
|IR NEC|Provided|[fbgentle1](https://github.com/fbgentle1)||
|IR RC-5|Provided|[fbgentle1](https://github.com/fbgentle1)||
|IR RC-6|Missing|||
|Sony IR (SIRC)|Provided|[fbgentle1](https://github.com/fbgentle1)||
|Timing jitter calculation|Missing|||
|Joint Test Action Group (IEEE 1149.1)|Missing|||
|NXP LFAST interface|Missing|||
|Low Pin Count|Missing|||
|SEGA Maple bus|Missing|||
|Intel MCS-48|Missing|||
|Management Data Input/Output|Missing|||
|Microwire|Provided|[fbgentle1](https://github.com/fbgentle1)||
|Miller encoding|Missing|||
|Morse code|Provided|[fbgentle1](https://github.com/fbgentle1)||
|Interpret bit patters as numbers or state enums|Missing|||
|1-Wire serial communication bus (link layer)|Provided|[fbgentle1](https://github.com/fbgentle1)||
|On-off keying|Missing|||
|Parallel sync bus|Missing|||
|Padded Jittering Data Link|Missing|||
|PS/2|Missing|||
|Pulse-width modulation|Provided|[DeeJay](https://github.com/DeeJay) & [fbgentle1](https://github.com/fbgentle1)||
|Qi charger protocol|Missing|||
|Remote control encoder|Missing|||
|RGB LED string decoder (WS281x)|Provided|[DeeJay](https://github.com/DeeJay)||
|SAE J1850 VPW.|Missing|||
|Siemens SDA 2506-5|Missing|||
|Secure Digital card (SD mode)|Missing|||
|Texas Instruments SDQ|Missing|||
|7-segment display|Missing|||
|Signature analysis|Missing|||
|SLE44xx memory card|Missing|||
|Sony/Philips Digital Interface Format|Missing|||
|Serial Peripheral Interface|Provided|[Dr. Gusman](https://github.com/gusmanb)|Only mode (0,0) tested, more mode captures required|
|Sitronix ST7735|Missing|||
|Stepper motor position / speed|Provided|[fbgentle1](https://github.com/fbgentle1)||
|Serial Wire Debug|Missing|||
|STM8 SWIM bus|Missing|||
|RFID T55xx|Missing|||
|Timing calculation with frequency and averaging|Missing|||
|Texas Instruments TLC5620|Missing|||
|Universal Asynchronous Receiver/Transmitter|Provided|[Dr. Gusman](https://github.com/gusmanb)||
|USB Power Delivery|Missing|||
|Universal Serial Bus (LS/FS) signalling|Provided|[mengstr](https://github.com/mengstr)|Low speed and full speed provided|
|Wiegand interface|Missing|||
|Zilog Z80 CPU|Provided|[fbgentle1](https://github.com/fbgentle1)||

Once the ammount of decoders tested is enough to have confidence that the system is more or less stable I will release a beta to the users that request it. I will also announce it here.

The faster we get the captures the sooner that I will be able to release the version, so if you can provide anyu capture for one of the untested decoders, please contribute 😄 

Thanks all!

**Comment by fbgentle1** [comment]
Hope this is what you are after.
I had to put a txt extension on the sigrok files
Cheers Bruce

![RC5_Code](https://github.com/user-attachments/assets/bb35e850-009d-4f99-9e77-63d151c81555)
![RC5_Code](https://github.com/user-attachments/assets/a0515a76-c87f-4a79-8fb6-029efc648fab)
[RC5_Code.sr.txt](https://github.com/user-attachments/files/16657180/RC5_Code.sr.txt)
[RC5_Code.pvs.txt](https://github.com/user-attachments/files/16657181/RC5_Code.pvs.txt)

**Reply by gusmanb** [reply]
Hi!

Thanks, what I need is the capture done with LogicAnalyzer, not with sigrok :)

**Comment by fbgentle1** [comment]
Hi,
Hopefully this file is better :-)
Here is an RC5 capture once again I added .txt to the lac file so it could be uploaded.

Cheers Bruce
[RC5_Code.lac.txt](https://github.com/user-attachments/files/16658804/RC5_Code.lac.txt)

**Reply by gusmanb** [reply]
Excellent! This is exactly what I need.

Thanks a lot :)

**Comment by fbgentle1** [comment]
Hi,
Here is an IIS trace for a Microphone with data on the Left channel and nothing on the Right.
Hope this is of some use.
Cheers Bruce
[IIS_Left_Channel.lac.txt](https://github.com/user-attachments/files/16671390/IIS_Left_Channel.lac.txt)

**Reply by gusmanb** [reply]
Of course it is 😄 

Thanks to these captures I'm fixing the implementation of the bridge, this capture for example raised a bug as I assumed that each annotation should be always in a row but the I2S decoder does not expose any row :)

Now it is fixed ;)

![imagen](https://github.com/user-attachments/assets/1d8a1437-31f8-4e68-97ee-7402d6b85201)

**Comment by fbgentle1** [comment]
Hi,
Here are a couple more IR protocols

Cheers Bruce
[IR_SIR_Video_8.lac.txt](https://github.com/user-attachments/files/16687571/IR_SIR_Video_8.lac.txt)
[IR_NEC_Extended_Protocol.lac.txt](https://github.com/user-attachments/files/16687572/IR_NEC_Extended_Protocol.lac.txt)

**Comment by fbgentle1** [comment]
Hi,

A few more for the collection.

Cheers Bruce
[DHT11.lac.txt](https://github.com/user-attachments/files/16703571/DHT11.lac.txt)
[OneWire_LinkLayer_DS1822.lac.txt](https://github.com/user-attachments/files/16703572/OneWire_LinkLayer_DS1822.lac.txt)

**Reply by gusmanb** [reply]
Excellen!

Thanks a lot :)
With your help things are evolving a lot faster.

**Comment by fbgentle1** [comment]
Hi,

Here is another one.

Cheers Bruce
[Microwire_93C46_Read_Addr21_Data64b4.lac.txt](https://github.com/user-attachments/files/16707894/Microwire_93C46_Read_Addr21_Data64b4.lac.txt)

**Comment by fbgentle1** [comment]
Hi,
Here is a Rotary encoder E6B2-CWZ6C 1000 pulses per revolution trace
Cheers Bruce
[Rotary_Encoder_1000pulse_rev_A_B_Z output.lac.txt](https://github.com/user-attachments/files/16720229/Rotary_Encoder_1000pulse_rev_A_B_Z.output.lac.txt)

**Comment by fbgentle1** [comment]
Hi,
A stepper motor controller trace this time
Cheers Bruce
[stepper_motor_controller.lac.txt](https://github.com/user-attachments/files/16733761/stepper_motor_controller.lac.txt)

**Comment by fbgentle1** [comment]
Hi,
Here is a morse code trace.
Cheers Bruce
[MorseCode_TimeUnit_0.01.lac.txt](https://github.com/user-attachments/files/16735146/MorseCode_TimeUnit_0.01.lac.txt)

**Comment by fbgentle1** [comment]
Hi,
Here is a PWM trace.
Cheers Bruce
[PWM.lac.txt](https://github.com/user-attachments/files/16737978/PWM.lac.txt)

**Reply by gusmanb** [reply]
Man, you are my hero, with all these captures  the system is mostly ready :)

If you want I can send you a preliminary version to test it, I would need to know OS type and device variant (pico/w/zero) 😄

**Reply by fbgentle1** [reply]
Hi Augustin,

That would be great, I am Windows 10 based and have made both PICO and PICO W devices.
It is a pleasure to help as you have put a lot of time and effort into this project which benefits many hobbyists.

Cheers Bruce

**Reply by gusmanb** [reply]
Here it is: https://mega.nz/file/CPxnVIzJ#kmvk6tJaLVad2T17vGAXL30eoVOFeaJZygsTFYIbZXc

I have included a small txt with info on how to get it working.

Hope you like it (and don't find too many bugs 😆)

**Reply by fbgentle1** [reply]
Thanks, 
I have downloaded it, will let you know how I go once it is setup and running.
Cheers Bruce

**Reply by fbgentle1** [reply]
Hi,
I have been trying a few of the decoders and they all work fine so far and look great.
I did have one issue with the Pico W board where it wouldn't capture any channel above 6 but I think it was to do with my system as reloading the firmware has fixed the issue.
I will continue testing but great work
Thanks Bruce

**Reply by gusmanb** [reply]
Perfect, thanks for the report. I have solved some bugs that I found that may be related to what you saw on the W, is it possible that you left a channel "empty"? a hole? There was a bug that caused problems in these cases.

I will send you this weekend a new version with all the fixes and also new capabilities (now you can hide/show the channels and the tooltips with all the info are ready, also I have changed the color parlette to something that I think is more readable).

Cheers!

**Reply by fbgentle1** [reply]
Yes, I definitely had an empty channel between the captured channels.
I look forward to the update.
Cheers Bruce

**Comment by sslupsky** [comment]
Have you had a look at the ngscopeclient project?  Might be of interest to you as it appears to be actively maintained.
https://www.ngscopeclient.org
https://github.com/ngscopeclient

**Reply by gusmanb** [reply]
I think that with V6.0 everyone is going to be more than happy :)

I managed to get to work the full decoding system from Sigrok even the stacking, improved the rendering speed and a lot of new functionalities :)

![imagen](https://github.com/user-attachments/assets/67276cd7-5450-4638-b4b3-5b68ce4cc280)

**Comment by mengstr** [comment]
A KA and a standard packet sequence from a LS USB from a mouse.  It's only sampled in 8Msps, but that should be enough to decode properly.

[USB-LowSpeed.lac.txt](https://github.com/user-attachments/files/16741729/USB-LowSpeed.lac.txt)

**Comment by mengstr** [comment]
USB Full Speed captured at 100Msps from an firmware update of a Black Magic Probe

[USB-HighSpeed.lac.txt](https://github.com/user-attachments/files/16754108/USB-HighSpeed.lac.txt)

**Reply by gusmanb** [reply]
Perfect, this one works much better :D

![imagen](https://github.com/user-attachments/assets/f9e0eb48-4633-4857-9f9b-4af5d7d71cd3)

**Comment by mengstr** [comment]
There's no smartcard ISO 7816 (wired) entry in the table above, but sigrok do have one - https://github.com/svenso/sigrok_iso7816

It's very hard to capture any decent amount of data considering the standard clock/data ratio is 372 clocks per databit. The reader I used has a clock at approx 5MHz so I sampled this file at 25Msps.  Hopefully there's enough data in the ADPU to have the decoder at least start doing its job.

[ISO7816.lac.txt](https://github.com/user-attachments/files/16762758/ISO7816.lac.txt)

**Reply by gusmanb** [reply]
The list is extracted from the included decoders on the PulseView distro, that particular one is not included.

In any case I added it manually and it seems to be working :)

![imagen](https://github.com/user-attachments/assets/469b7bb4-ff5c-4030-927c-441cbbb6a514)


Also, I have already implemented the pinning of the preview and the info of the current pulse :D

**Reply by mengstr** [reply]
Nice!  This is definitely a case where a much larger memory buffer is needed, or a synchronous clock (but that would only get 5x more samples which is probably not enough to capture the entire ATR message).

**Comment by fbgentle1** [comment]
Hi,
Here is a Z80 capture, it doesn't have the address lines captured but they are optional.
The decoding was correct and couldn't see any issues but thought it may be of interest as it is a little more complex capture.
Cheers Bruce

[Z80_CPM.lac.txt](https://github.com/user-attachments/files/16792075/Z80_CPM.lac.txt)

**Reply by fbgentle1** [reply]
Hi,

Something a little different.
I thought I would make a Zero version of the analyzer so took a clone of V6.0.
I could compile the firmware successfully but the USB wouldn't work.
After much hunting around I found that the delay_cycles4 routine appeared to kill the usb.
Looking at the disassembly everything was correct and I could see the inline code compiled correctly, and therefore couldn't see any reason for the issue.
I changed the delay to simply use busy_wait_at_least_cycles(loops*2) and not inline and that works and doesn't kill the usb.
Have you seen this issue before or with version 6?

Cheers Bruce

**Reply by gusmanb** [reply]
Wait, it is not going to work, I'm modifying the protocol and the driver and it has not been tested, to be honest I'm amazed that it even compiled properly 😄 .
I expect to test and correct it this weekend, if you want once I complete it I can send you a new version of the software and the firmware.

I need to start following a better branching naming convention to make these more clear, sorry.

**Reply by fbgentle1** [reply]
No problems, that would be great.
I will wait for the release. :-)

**Reply by gusmanb** [reply]
Ok, as promised here is the version, and it comes with a really big update.

https://mega.nz/file/aTZgRTBT#8tL-JOV_PvWrcu0kQvRJE9ni11GeVcZBdSe2XdRYkr4

I have included the software and the firmware for the Pico and the Zero, beware that I have reworked the protocol and the old firmwares will not work (the W still not completed, so if you plan to use it you will need the previous software).

And now, the big update, I've managed to get it stable running at 400Mhz, that means that the analyzer now is capable of capturing up to 200Ms/s 😄 

One of the new features that is already implemented (the burst measurement) made me to move the build to RAM only as reading from the flash causes delays that break the measurements, and the problems that I had on the past doing this extreme overclocking were with the flash (not sure why as in theory the QMI should be reconfigured automatically based on the clock speed... no idea), and now as it is no more in it the analyzer works properly 🕺 .

Beware that this is an extreme overclocking and I had to overvolt the core to 1.35v to get it stable, so I'm not responsible if the CPU burns out, yo do it on your own 😅 

There is a sety of two firmwares per device a regular one that captures up to 100Mhz and other named TURBO that captures up to 200Mhz. The GUI will detect the max speed and will apply the needed restrictions.

I have tested it in three picos and one zero and I had no problems, but each chip is a world and who knows if it may burn... In any case I would recommend to put a little heat sink to the RP2040 if you plan to use it for long periods (that's what I did).

Have fun!

**Reply by fbgentle1** [reply]
Hi,
Great work, the updates sound great and can't wait to try them.
I have put a heatsink on the pico and am now ready to _destroy_ I mean test it 😅

![20240908_084607sm](https://github.com/user-attachments/assets/5a7644fc-e19d-42b7-8b54-ef48a5aa2f2a)

I will let you know how things go.

Thanks Bruce

**Reply by fbgentle1** [reply]
Hi,
Just having a quick look with the Zero non Turbo and seem to get some funny issues when setting the samples (see below)
![image](https://github.com/user-attachments/assets/5425f67a-8fed-4f00-bf01-54614e5e2df3)
![image](https://github.com/user-attachments/assets/4b705ff2-53e7-4fa3-9f9e-bbfd0e9245ee)

The second issue is also around samples. The calculation or limit appears to be incorrect
![image](https://github.com/user-attachments/assets/78cb61e2-beb5-4c9e-8e57-4330dd5c9859)

Cheers Bruce

**Reply by gusmanb** [reply]
Yesah, there was an error in the driver were I compared the max presamples with the specified postsamples in the driver... 😆 

Here is the driver DLL, replacing it should be enough.

[SharedDriver.zip](https://github.com/user-attachments/files/16920027/SharedDriver.zip)

About the second error, I can't see if you enabled the burst mode, I assume that you did, in that case each burst is of 90,000 samples, two bursts would be 180,000 samples and that exceeds the buffer size.

Cheers.

**Reply by fbgentle1** [reply]
Ha,
These things happen and you only find out just after you release it !
Also that's why it is good to get other people to test 😅
The new driver dll seems to have fixed the issue.

And yes you are correct about the burst mode...my bad

Thanks Bruce

**Reply by DeeJay** [reply]
v6 preview, with updated SharedDriver. WIndows10 workstation, and 'classic' original Pico - nonTurbo. USB-Serial connection.

Basic signal acquisition and display working as expected.

No decoding possible.

At startup a warning appears -
`Cannot load Sigrok decoders. Make sure Python is installed on your computer. If, despite being installed, you still have problems, you can specify the path to the Python library in "python.cfg".`

I have python installed -
```
C:\Users\David>python
Python 3.9.2 (tags/v3.9.2:1a79785, Feb 19 2021, 13:44:55) [MSC v.1928 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>>
```

But I have no python.cfg in my Software directory. If I create it, what should be the format of its content?

**Reply by gusmanb** [reply]
Hi. The file must be a single line with the full path to the python3x.dll, where X is the version.

Something like this:
`C:\Program Files\Python39\Python39.dll`

Could you do me a favor? 
Execute these two things:

First, directly from the console at the folder where you have decompressed the analyzer software, this: `python -c "import sys; ver = str(sys.version_info.major) + str(sys.version_info.minor); print(ver)"`

It should return the version number of Python, in your case should return "39"

If it works then open a Python console and execute this script:

```
import sys;
import os.path as op;
ver = str(sys.version_info.major) + str(sys.version_info.minor);
path = sys.exec_prefix;
print(op.join(path, 'Python' + ver + '.dll'));
```

This is the scripts that I use to detect where Python is installed, it should yield the exact path and name of the DLL, if it works and the DLL has a different name please add also the real name of the DLL.

For example, I have python 3.9.13 installed and the second script yields "C:\Program Files\Python39\Python39.dll", it should be the same for you as you are in 3.9.2.

If by any chance using the cfg file does not work and all the previous tests are correct I will compile a version with debug output to validate why it is not finding the python installation.

Thanks in advance.

**Reply by DeeJay** [reply]
_{A fuller reply got lost somewhere. A quick summary...}_

tl;dr - 
With C:\Users\David\AppData\Local\Programs\Python\Python39\Python39.dll
as the content of python.cfg the top panel of the right-side of the LA screen is populated with Decoders

Using the UART decoder I am able to recover the ascii text in the messages being sent by the external signal generator. 

All good!


Background
```
C:\Users\David\Downloads\LA_60_Preview2\Software>python
Python 3.9.2 (tags/v3.9.2:1a79785, Feb 19 2021, 13:44:55) [MSC v.1928 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> import os.path as op
>>> ver = str(sys.version_info.major) + str(sys.version_info.minor)
>>> path = sys.exec_prefix
>>> print(op.join(path, 'Python' + ver + '.dll'))
C:\Users\David\AppData\Local\Programs\Python\Python39\Python39.dll
>>>

```

**Reply by gusmanb** [reply]
Hmmmm, really strange, the script is returning the correct path but it does not find the library automatically...

Could you try this executable and dll please? Preserve the old ones as this is built in debug mode and will generate a log each time you execute it.

[Debug_Executable.zip](https://github.com/user-attachments/files/16933080/Debug_Executable.zip)

Rename the "python.cfg", "LogicAnalyzer.dll" and "LogicAnalyzer.exe" files to something else, extract the files in the zip to the analyzer software folder and run it from the console, it will create a "log.txt" file with detailed info on the python system initialization, once it is done you can revert back to the old exe and paste the content or upload the file here.

Thanks in advance 😄

**Reply by DeeJay** [reply]
Sorry to report that the LogicAnalyzer.exe in the Debug archive does not run at all. When invoked from the Command Prompt CLI it just silently returns to the prompt. No GUI app appears, no log or config files are created.

```
C:\Users\David\Downloads\LA_60_Preview2\Software>
C:\Users\David\Downloads\LA_60_Preview2\Software>dir Logic*
 Volume in drive C is Windows
 Volume Serial Number is DE90-EBAE

 Directory of C:\Users\David\Downloads\LA_60_Preview2\Software

09/09/2024  11:09           113,086 LogicAnalyzer.deps.json
09/09/2024  18:19         6,578,688 LogicAnalyzer.dll
09/09/2024  18:19         1,098,240 LogicAnalyzer.exe
09/09/2024  11:09           107,620 LogicAnalyzer.pdb
09/09/2024  11:09               448 LogicAnalyzer.runtimeconfig.json
               5 File(s)      7,898,082 bytes
               0 Dir(s)  32,366,989,312 bytes free

C:\Users\David\Downloads\LA_60_Preview2\Software>
C:\Users\David\Downloads\LA_60_Preview2\Software>LogicAnalyzer.exe

C:\Users\David\Downloads\LA_60_Preview2\Software>
C:\Users\David\Downloads\LA_60_Preview2\Software>dir log.*
 Volume in drive C is Windows
 Volume Serial Number is DE90-EBAE

 Directory of C:\Users\David\Downloads\LA_60_Preview2\Software

File Not Found

C:\Users\David\Downloads\LA_60_Preview2\Software>dir python.*
 Volume in drive C is Windows
 Volume Serial Number is DE90-EBAE

 Directory of C:\Users\David\Downloads\LA_60_Preview2\Software

09/09/2024  11:09           431,616 Python.Runtime.dll
               1 File(s)        431,616 bytes
               0 Dir(s)  32,322,842,624 bytes free

C:\Users\David\Downloads\LA_60_Preview2\Software>

```

**Reply by gusmanb** [reply]
Hmmm, that's a bit odd...

Here is the full build, drop it in another folder and it should work, I tested it before compressing it.

https://mega.nz/file/vHxRwZjL#5uwFn4H6lIFXliMzExHWwNp5uAxLIt5-kryxbAAV34Y

Just by any chance, do you have any kind of high security implemented in your system? Like disk quotas or restricted permissions?

Cheers.

**Reply by DeeJay** [reply]
Thanks for your attention. I am not in urgent need of a solution, but am happy to help you debug this in case it is something that might affect other potential users of v6.0

My Windows version is "Windows 10 Home" v22H2 build 19045.4780.  It is my own home Desktop machine. I am the admin. It uses traditional local default user + password challenge authentication. I don't recall ever implementing quotas or restrictive permissions.

Your debug build does execute as expected. It still gives the warning about Sigrok decoders, and no decoders are loaded.

 **log.txt** does get created -

```
C:\Users\David\Downloads\debug_build\win-x64>dir log.txt
 Volume in drive C is Windows
 Volume Serial Number is DE90-EBAE

 Directory of C:\Users\David\Downloads\debug_build\win-x64

10/09/2024  08:05               716 log.txt
               1 File(s)            716 bytes
               0 Dir(s)  32,022,818,816 bytes free

C:\Users\David\Downloads\debug_build\win-x64>type log.txt
10/09/2024 - 10/09/2024: Initializing python system...
10/09/2024 - 10/09/2024: Finding python library...
10/09/2024 - 10/09/2024: Testing python3 interpreter...
10/09/2024 - 10/09/2024: python3 interpreter found.
10/09/2024 - 10/09/2024: Python version is not 3.x, found  aborting.
10/09/2024 - 10/09/2024: Python library not found, aborting startup.
10/09/2024 - 10/09/2024: Initializing python system...
10/09/2024 - 10/09/2024: Finding python library...
10/09/2024 - 10/09/2024: Testing python3 interpreter...
10/09/2024 - 10/09/2024: python3 interpreter found.
10/09/2024 - 10/09/2024: Python version is not 3.x, found  aborting.
10/09/2024 - 10/09/2024: Python library not found, aborting startup.

C:\Users\David\Downloads\debug_build\win-x64>python
Python 3.9.2 (tags/v3.9.2:1a79785, Feb 19 2021, 13:44:55) [MSC v.1928 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>>
>>>

```


It feels as though there is another python installation laying around that is being detected by your tests but not by the Command Prompt CLI?

**Reply by gusmanb** [reply]
Yes, it seems you have multiple python installations.

Try this from the command line: `python3 -c "import sys; ver = str(sys.version_info.major) + str(sys.version_info.minor); print(ver)"`. If it works please paste the result of the command.

Note the difference, instead of running `python` you're running `python3`.

And then, from the console run directly `python3` and then the script to get the path:

```
import sys;
import os.path as op;
ver = str(sys.version_info.major) + str(sys.version_info.minor);
path = sys.exec_prefix;
print(op.join(path, 'Python' + ver + '.dll'));
```

Sorry for bothering you with these tests but this is very valuable information to get the python detection system working as smooth as possible 😄 

Thanks!

**Reply by DeeJay** [reply]
I think you are homing in on the issue.

I cannot run the tests you propose. 

From a Command Prompt session -

Check **python**
```
Microsoft Windows [Version 10.0.19045.4780]
(c) Microsoft Corporation. All rights reserved.

C:\Users\David>python
Python 3.9.2 (tags/v3.9.2:1a79785, Feb 19 2021, 13:44:55) [MSC v.1928 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>>

```

Check **python3**
```
Microsoft Windows [Version 10.0.19045.4780]
(c) Microsoft Corporation. All rights reserved.

C:\Users\David>python3

C:\Users\David>
```

The latter fails to invoke a python interpreter and brings up a Microsoft Store window proposing the installation of Python 3.12.6




Separately, I have done a Windows file Explorer search for 'python3*.dll' and found 6 matches.

2 are for the version in AppData\Local\Programs\Python:  ...\Python39\python3.dll and  ...\Python39\python39.dll

2 are related to Thonny - a python-oriented IDE that provides its own python interpreter - python310

2 are related to LibreOffice - an MS Office 'clone' - python35


But I suspect it is the simple fact that 'python3' does not map to a python interpreter that is the problem for LAv6.0

I don't know how I innocently created this situation. But maybe other potential users will have done the same thing.

**Reply by gusmanb** [reply]
I think is not you who caused this situation but MS, I bet you they added some kind of hard link to the store if you invoke python3 and is not installed with that name...

But now it's clear what is happening and I know how to solve it, instead of once an executable that matches is found, stop trying the others and fail, if is not correct I'm going to change the procedure to do the full detection on both possibilities, this should solve this problem.

Once I get the new detection routine ready I will upload it here.

Thanks a lot for your help 😄

**Reply by gusmanb** [reply]
Here it is:

https://mega.nz/file/yOYFyJqD#XUafuycj8lXLk3lilIS4rCwb00AsYSp-Xfi8GxG7nnY

This one will test both interpreter names and even in the case that there are two valid Python 3 installs it will choose the best one that can find.

Also, the log is now called "PythonInitLog.txt", I think that until this is well tested I'm going to leave it on as it is really useful.

Cheers.

**Reply by DeeJay** [reply]
Starts up with no warning dialog and with Sigrok decoders loaded into a panel at the Right side of the window.

Log file shows -
```
10/09/2024 11:35 -> Initializing python system...
10/09/2024 11:35 -> Initializing python installation detection...
10/09/2024 11:35 -> Target Interpreter: python,  Initializing Python Installation Detection...
10/09/2024 11:35 -> Target Interpreter: python,  Python version found: 39
10/09/2024 11:35 -> Target Interpreter: python,  Python version parsed: 3.9
10/09/2024 11:35 -> Target Interpreter: python,  Finding python library on Windows...
10/09/2024 11:35 -> Target Interpreter: python,  Script output: C:\Users\David\AppData\Local\Programs\Python\Python39\Python39.dll
10/09/2024 11:35 -> Target Interpreter: python,  Final python path: C:\Users\David\AppData\Local\Programs\Python\Python39\Python39.dll
10/09/2024 11:35 -> Target Interpreter: python,  Python path found: C:\Users\David\AppData\Local\Programs\Python\Python39\Python39.dll
10/09/2024 11:35 -> Target Interpreter: python3,  Initializing Python Installation Detection...
10/09/2024 11:35 -> Target Interpreter: python3,  Python version not found, aborting.
10/09/2024 11:35 -> Valid installations found: 1
10/09/2024 11:35 -> Selected version: 3.9
10/09/2024 11:35 -> Stablished path: C:\Users\David\AppData\Local\Programs\Python\Python39\Python39.dll
10/09/2024 11:35 -> Initializing decoders...
10/09/2024 11:35 -> Python initialization completed.
```

Tested with contents of unpacked archive only. I did NOT copy across the python.cfg file, and one was not present after the app exited.

**Reply by gusmanb** [reply]
Excellent.

Thanks a lot for your help 😄

**Comment by DeeJay** [comment]
Returning to the topic of getting trial data for device decoding...

RGB LED string decoder (WS281x), captured with LA60pre. (The Sigrok decoder in LA60pre is able to interpret this capture.)
(I can provide a v5.1 version of the capture on request.)

[RGB_LED_WS281x.zip](https://github.com/user-attachments/files/16943165/RGB_LED_WS281x.zip)

The pattern sequence is intended to be [OFF, OFF, State_A, State_B, State_C, State_C, State_B, State_A, OFF, OFF] . Part of this sequence is correctly shown in the capture.

**Comment by DeeJay** [comment]
Pulse-width modulation

I think you have had one offer, but it is still marked 'missing.'

Here is another offering: 4 channels, all 100Hz base frequency, with different Duty Cycles as implied by the Channel name tags.

Capture performed by released version 5.1 of LA. The LA60pre2 code can Open this capture and correctly decode it.

[PWM_duty_cycles.zip](https://github.com/user-attachments/files/16947851/PWM_duty_cycles.zip)

**Comment by henrygab** [comment]
There are many, many traces of RFID (EM4100, EM4205/EM4305 included) here:

https://github.com/RfidResearchGroup/proxmark3/tree/master/traces

The `.pm3` files are all single-channel traces:
* One sample per line
* Each line is ASCII, encoding a single digit, typically in range [127,-127].
* For EM41xx/EM42xx/EM43xx, the sample rate is 125kHz.

<details><summary>Direct links to relevant traces</summary>
<p>



Can you convert those traces to what you need for RFID (EM4100, EM4205/EM4305) samples?

* https://github.com/RfidResearchGroup/proxmark3/tree/master/traces/lf_ATA5577_em410x.pm3
* https://github.com/RfidResearchGroup/proxmark3/tree/master/traces/lf_EM4102-1.pm3
* https://github.com/RfidResearchGroup/proxmark3/tree/master/traces/lf_EM4102-2.pm3
* https://github.com/RfidResearchGroup/proxmark3/tree/master/traces/lf_EM4102-3.pm3
* https://github.com/RfidResearchGroup/proxmark3/tree/master/traces/lf_EM4102-clamshell.pm3
* https://github.com/RfidResearchGroup/proxmark3/tree/master/traces/lf_EM4102-fob.pm3
* https://github.com/RfidResearchGroup/proxmark3/tree/master/traces/lf_EM4102-thin.pm3
* https://github.com/RfidResearchGroup/proxmark3/tree/master/traces/lf_EM4305_fdxa_destron.pm3
* https://github.com/RfidResearchGroup/proxmark3/tree/master/traces/lf_EM4x05.pm3
* https://github.com/RfidResearchGroup/proxmark3/tree/master/traces/lf_EM4x50.pm3

</p>
</details>

**Reply by gusmanb** [reply]
Will check it, thanks for the info.

**Reply by henrygab** [reply]
There are a number of traces specifically showing various modulations:

Biphase, Manchester, FSK (frequency-shift keying), PSK (phase-shift keying), NRZ, ...
filename | description
-----|-----
`lf_Q5_mod-*` | Q5 configured to emit 00 01 02 03 04 05 06 07 08 09 0A 0B under various modulation schemes

If you have a script to convert `.PM3` files to your preferred format, that should give quite a nice selection of patterns for testing decoders.


Note:  In past, I've noticed sometimes the ASCII value in `.pm3` files slightly exceeds the [127,-127] range; I recommend checking if the values are >95% in that range and 100% within the range [135,-135], and if so, cap the values to range [127,-127].

**Reply by henrygab** [reply]
_**[FYI update (no response expected)]**_

I've submitted [PR 2518](https://github.com/RfidResearchGroup/proxmark3/pull/2518) to that proxmark repository, which was merged.
Now all the `.pm3` files keep their values in the range `[127,-128]`, so each line can be converted to a single signed byte value.

**Comment by omicronns** [comment]
Here is SWD (Serial Wire Debug) sample, captured on STM32F0308-DISCO, when connecting with the target using STM32CubeProgrammer.
[swd_sample1.zip](https://github.com/user-attachments/files/17483521/swd_sample1.zip)

**Comment by firew4lker** [comment]
Here is a CAN bus capture (ISO 15765-4). 500 kbps 11 bit address. 

Data captured with a Pico scope:
![Screenshot 2024-12-29 120718](https://github.com/user-attachments/assets/eef67d9c-4666-4df9-bc4e-d139ed26b3d5)

Data Captured with Logic Analyzer. It seems OK!

![Screenshot 2024-12-29 122326](https://github.com/user-attachments/assets/a6bc3bb9-e899-4ce8-89b1-d8829fae035b)


[CAN-BUS.zip](https://github.com/user-attachments/files/18269103/CAN-BUS.zip)

**Comment by firew4lker** [comment]
SAE J1850 VPW also seems to be OK!

Data send: 0x01, 0x02, 0xAA

Data captured:
![Screenshot 2024-12-29 153337](https://github.com/user-attachments/assets/741bea71-8d00-4b1b-b492-1142d723bfd9)

[J1850_VPW.zip](https://github.com/user-attachments/files/18269759/J1850_VPW.zip)
